pkg/karmadactl/addons/init/enable_option.go

```diff
@@ -0,0 +1,157 @@
+	"k8s.io/klog/v2"
+	aggregator "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset"
+	"k8s.io/utils/strings/slices"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	karmadaclientset "github.com/karmada-io/karmada/pkg/generated/clientset/versioned"
+	"github.com/karmada-io/karmada/pkg/karmadactl/cmdinit/utils"
+	"github.com/karmada-io/karmada/pkg/util/gclient"
+)
+
+// CommandAddonsEnableOption options for addons list.
+type CommandAddonsEnableOption struct {
+	// KubeConfig holds host cluster KUBECONFIG file path.
+	KubeConfig string
+	Context    string
+
+	// KubeConfig holds karmada control plane KUBECONFIG file path.
+	KarmadaConfig  string
+	KarmadaContext string
+
+	// Namespace holds the namespace where Karmada components intalled
+	Namespace string
+
+	KarmadaSearchImage string
+
+	KarmadaSearchReplicas int32
+
+	KarmadaDeschedulerImage string
+
+	KarmadaDeschedulerReplicas int32
+
+	KarmadaEstimatorImage string
+
+	KarmadaEstimatorReplicas int32
+
+	KubeClientSet *kubernetes.Clientset
+
+	KarmadaKubeClientSet *kubernetes.Clientset
+
+	KarmadaClientSet *karmadaclientset.Clientset
+
+	KarmadaAggregatorClientSet *aggregator.Clientset
+
+	ControlPlaneClient client.Client
+
+	WaitPodReadyTimeout int
+
+	WaitAPIServiceReadyTimeout int
+
+	Cluster string
+
+	MemberConfig string
+
+	MemberContext string
+}
+
+// Complete the conditions required to be able to run enable.
+func (o *CommandAddonsEnableOption) Complete() error {
+	if o.KubeConfig == "" {
+		env := os.Getenv("KUBECONFIG")
+		if env != "" {
+			o.KubeConfig = env
+		} else {
+			o.KubeConfig = filepath.Join(homedir.HomeDir(), ".kube", "config")
+		}
+	}
+
+	restConfig, err := utils.RestConfig(o.Context, o.KubeConfig)
+	if err != nil {
+		return err
+	}
+
+	o.KubeClientSet, err = utils.NewClientSet(restConfig)
+	if err != nil {
+		return err
+	}
+
+	karmadaRestConfig, err := utils.RestConfig(o.KarmadaContext, o.KarmadaConfig)
+	if err != nil {
+		return err
+	}
+
+	o.KarmadaKubeClientSet, err = utils.NewClientSet(karmadaRestConfig)
+	if err != nil {
+		return err
+	}
+
+	o.KarmadaClientSet, err = utils.NewKarmadaClientSet(karmadaRestConfig)
+	if err != nil {
+		return err
+	}
+
+	o.KarmadaAggregatorClientSet, err = utils.NewAPIRegistrationClient(karmadaRestConfig)
+	if err != nil {
+		return err
+	}
+
+	o.ControlPlaneClient = gclient.NewForConfigOrDie(karmadaRestConfig)
+
+	return nil
+}
+
+// Validate Check that there are enough conditions to run the enable.
+func (o *CommandAddonsEnableOption) Validate(args []string) error {
+	secretClient := o.KubeClientSet.CoreV1().Secrets(o.Namespace)
+	_, err := secretClient.Get(context.TODO(), "kubeconfig", metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return fmt.Errorf("secrets `kubeconfig` is not found in namespace %s, please execute karmadactl init to deploy karmada first", o.Namespace)
+		}
+	}
+
+	if slices.Contains(args, EstimatorResourceName) && o.Cluster == "" {
```

Comment:
Dose it need to check `o.MemberKubeconfig` and `o.MemberContext`?

Because when I execute

```
[root@master67 karmada]# ./karmadactl addons enable karmada-scheduler-estimator -C member2 --karmada-kubeconfig /root/.kube/config --karmada-context karmada-apiserver --kubeconfig /root/.kube/config --context karmada-host
```

It was deployed successfully, but the `logs` is warning.

```
[root@master67 ~]# khost logs -n karmada-system karmada-scheduler-estimator-member1-8548dfc64-5bk9d
I0720 05:54:20.557295       1 scheduler-estimator.go:66] karmada-scheduler-estimator version: version.Info{GitVersion:"v1.2.0", GitCommit:"cc74a423222ccf1b6896d7fa8c314544a24cade7", GitTreeState:"clean", BuildDate:"2022-05-28T15:43:04Z", GoVersion:"go1.17.10", Compiler:"gc", Platform:"linux/amd64"}
I0720 05:54:20.569931       1 server.go:141] Starting karmada cluster(member1) accurate scheduler estimator
W0720 05:54:20.598822       1 reflector.go:324] k8s.io/client-go/informers/factory.go:134: failed to list *v1.ReplicaSet: replicasets.apps is forbidden: User "system:serviceaccount:karmada-system:default" cannot list resource "replicasets" in API group "apps" at the cluster scope
E0720 05:54:20.598905       1 reflector.go:138] k8s.io/client-go/informers/factory.go:134: Failed to watch *v1.ReplicaSet: failed to list *v1.ReplicaSet: replicasets.apps is forbidden: User "system:serviceaccount:karmada-system:default" cannot list resource "replicasets" in API group "apps" at the cluster scope
W0720 05:54:20.598983       1 reflector.go:324] k8s.io/client-go/informers/factory.go:134: failed to list *v1.Node: nodes is forbidden: User "system:serviceaccount:karmada-system:default" cannot list resource "nodes" in API group "" at the cluster scope
E0720 05:54:20.598998       1 reflector.go:138] k8s.io/client-go/informers/factory.go:134: Failed to watch *v1.Node: failed to list *v1.Node: nodes is forbidden: User "system:serviceaccount:karmada-system:default" cannot list resource "nodes" in API group "" at the cluster scope
W0720 05:54:20.599074       1 reflector.go:324] k8s.io/client-go/informers/factory.go:134: failed to list *v1.Pod: pods is forbidden: User "system:serviceaccount:karmada-system:default" cannot list resource "pods" in API group "" at the cluster scope
```